src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,82 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+
+    static {
+        interfaceImplementations =
+                Map.of(FileReaderService.class, FileReaderServiceImpl.class,
+                        ProductParser.class, ProductParserImpl.class,
+                        ProductService.class, ProductServiceImpl.class
+                );
+    }
+
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> interfaceImplementations;
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+
+        Object clazzImplementationInstance = null;
+        Class<?> clazzImplementation = findImplementation(interfaceClazz);
+        if (!clazzImplementation.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class implementation: "
+                    + clazzImplementation.getName()
+                    + "doesn't have 'Component' annotation ");
+        }
+        Field[] fields = clazzImplementation.getFields();
```

Comment:
You should use clazz.getDeclaredFields() instead of clazz.getFields() to also include private fields in your search, as getFields() only returns public fields including inherited ones.